firebase_to_xml/firebase_to_xml.py

```diff
@@ -0,0 +1,221 @@
+                        'organization':{
+                            'name': x.get('orgName'),
+                            'url': x.get('orgURL'),
+                            'address': x.get('orgAdress'),
+                            'city': x.get('orgCity'),
+                            'country': x.get('orgCountry'),
+                            'email': x.get('orgEmail')
+                        },
+                        'individual': {
+                            'name': x.get('indName'),
+                            'position': x.get('indPosition'),
+                            'email': x.get('indEmail'),
+                            # country:
+                        }
+                    } for x in r.get('contacts', [])
+                ],
+
+                'distribution': [
+                    {
+                        'url': [x.get('url')],
+                        'name': x.get('name'),
+                        'description': x.get('description'),
+                        #   en: pdf description in English
+                        #   fr: pdf description in French
+                    } for x in r.get('distribution', [])
+                ],
+                'platform': {
+                    'name': r.get('platformName'),
+                    'role': r.get('platformRole'),
+                    # authority: platform_authority
+                    'id': r.get('platformID'),
+                    'description': r.get('platformDescription'),
+                    'instruments': r.get('instruments')
+                    # - id: 123
+                    #   manufacturer: manufacturer en 1
+                    #   version: A1.53
+                    #   type:
+                    #     en: type en 1
+                    #     fr: type fr 1
+                    #   description:
+                    #     en: instrument description en 1
+                    #     fr: instrument description fr 1
+                }
+            }
+
+            if r.get('created') is not None:
+                yDict['identification']['dates']['creation'] = r.get('created')
+            if r.get('published') is not None:
+                yDict['identification']['dates']['publication'] = r.get(
+                    'published')
+            if r.get('revised') is not None:
+                yDict['identification']['dates']['revision'] = r.get('revised')
+
+            # check at least one date has been added otherwise skip this record
+            if not yDict.get('identification') or not yDict.get('identification').get('dates'):
+                raise Exception(
+                    'at least one entry in identification.dates is required')
+
+            # render xml template and write to file
+            xml = template.render({"record": yDict})
+            id = r.get('recordID')
+            name = r.get('title', {}).get(r.get('language'))
+            if name:
+                name = name.strip().lower().replace(' ', '_')
```

Comment:
might need to filter out more characters to sanitize the filename

Reply:
changed in most recent push